src/main/kotlin/org/jitsi/nlj/transform/node/outgoing/OutgoingStatisticsTracker.kt

```diff
@@ -15,82 +15,58 @@
  */
 package org.jitsi.nlj.transform.node.outgoing
 
-import org.jitsi.utils.stats.RateStatistics
 import org.jitsi.nlj.PacketInfo
+import org.jitsi.nlj.stats.PacketStreamStats
 import org.jitsi.nlj.transform.node.ObserverNode
 import org.jitsi.rtp.rtp.RtpPacket
-import java.time.Duration
 import java.util.concurrent.ConcurrentHashMap
 
 class OutgoingStatisticsTracker : ObserverNode("Outgoing statistics tracker") {
-    private val ssrcStats: MutableMap<Long, OutgoingSsrcStats> = ConcurrentHashMap()
-
     /**
-     * The bitrate in bits per seconds.
+     * Per-SSRC statistics
      */
-    private val bitrate = RateStatistics(Duration.ofSeconds(1).toMillis().toInt())
+    private val ssrcStats: MutableMap<Long, OutgoingSsrcStats> = ConcurrentHashMap()
     /**
-     * The packet rate in packets per second.
+     * The stats for all of the SSRCs combined.
      */
-    private val packetRate = RateStatistics(Duration.ofSeconds(1).toMillis().toInt(), 1000f)
-
-    private var bytesSent: Long = 0
-    private var packetsSent: Long = 0
+    private val combinedStats = PacketStreamStats()
 
     override fun observe(packetInfo: PacketInfo) {
-        val rtpPacket = packetInfo.packetAs<RtpPacket>()
-        val stats = ssrcStats.computeIfAbsent(rtpPacket.ssrc) {
-            OutgoingSsrcStats(rtpPacket.ssrc)
-        }
-        stats.packetSent(rtpPacket.length, rtpPacket.timestamp)
+        combinedStats.update(packetInfo.packet.length)
 
-        val now = System.currentTimeMillis()
-        val bytes = rtpPacket.length
-        bitrate.update(bytes, now)
-        packetRate.update(1, now)
-        bytesSent += bytes
-        packetsSent++
+        (packetInfo.packet as? RtpPacket) ?.let { rtpPacket ->
+            val stats = ssrcStats.computeIfAbsent(rtpPacket.ssrc) {
+                OutgoingSsrcStats(rtpPacket.ssrc)
+            }
+            stats.packetSent(rtpPacket.length, rtpPacket.timestamp)
+        }
     }
 
     fun getSnapshot(): OutgoingStatisticsSnapshot {
-        val now = System.currentTimeMillis()
         return OutgoingStatisticsSnapshot(
-            bitrate.getRate(now),
-            packetRate.getRate(now),
-            bytesSent,
-            packetsSent,
+            combinedStats.snapshot(),
             ssrcStats.map { (ssrc, stats) ->
                 Pair(ssrc, stats.getSnapshot())
             }.toMap()
         )
     }
 
+    /**
+     * Gets the combined stats (i.e. overall stats for sent packets).
+     */
+    fun getCombinedStatsSnapshot() = combinedStats.snapshot()
```

Comment:
i still think the name of this (and everything in this chain) should reference "packet stream stats".  reading `getCombinedStatsSnapshot` or `combinedStats` tells someone nothing about what these stats are, even if they _did_ know what `PacketStreamStats` were.